### ISSUE TITLE
The quality of generative images is lower than the paper's result?

### ISSUE THREAD

**Issue by zhangqianhui** [carpedm20/BEGAN-tensorflow#1]
Really?

**Comment by carpedm20** [comment]
I just received some comment on learning rate which was different from the code which makes the images in README.md. I'll update the result and authors will update and clarify implementation details near future.

**Comment by zhangqianhui** [comment]
ok.

**Comment by artcg** [comment]
You shouldnt expect to necessarily re-create same quality as the paper anyway since the paper uses a not released to public dataset

**Comment by carpedm20** [comment]
@artcg I thought they only used CelebA which is public. Anyway, thanks to your comment, I just found that they use totally different crop (more face focused) and it should be definitely helpful to reproduce the result more clearly.

**Comment by artcg** [comment]
Nope from the paper 'We use a dataset of 360k celebrity face images for training in place of CelebA' 

I.e. Instead of using the standard CelebA they are using this larger dataset with different cropping / alignment like you say

But that is a good point maybe different pre-processing on CelebA would improve results!

**Comment by carpedm20** [comment]
There was a `tf.slim` related mistake as discussed in #3 and now the results are way better as [this](https://github.com/carpedm20/BEGAN-tensorflow#generator-and-discriminator-outputs-after-58k-step). However, 128x128 image for CelebA doesn't goes well with the default hyper-parameters (only 64x64 trained well).

**Comment by surya501** [comment]
I tried this on the zappos dataset. Hence, I don't have a baseline comparison to compare against. However, *my* feeling is that the output could be better and definitely not at the same level as the faces in the paper. Some output for reference below. This is after 500k iterations over ~50 hours of training on GTX 1080
![test8_g_z](https://cloud.githubusercontent.com/assets/4899994/25097760/bc87659a-235a-11e7-95ea-fd8e4d96cf9d.png)
![test9_g_z](https://cloud.githubusercontent.com/assets/4899994/25097771/c2106200-235a-11e7-8c79-c79387c3729c.png)

**Comment by fwiffo** [comment]
That Zappos example looks like mode collapse. It's producing many examples of a small number of shoes. Different datasets require different tuning of hyperparameters so I would suggest lowering the learning rate or adjusting the size of z and/or h.

**Comment by surya501** [comment]
Thanks @fwiffo. Any rule of thumb/heuristics on which to try first... especially because the runtimes are long.

**Comment by fwiffo** [comment]
Start with learning rate. Reduce it by half, and/or decay it more frequently during training. Also, you don't need to train for that many steps to see if it's going to mode collapse; it should do that in the first 100k steps. Next try adjusting h. For faces the quality was significantly worse with 32, and somewhat worse with 128, but this probably depends on the diversity of the dataset. The size of z seems to have much less effect; anywhere from 32-1024 seemed similar in our tests. Started to see reduced diversity and quality below 32, but it may depend on the dataset.

Smaller datasets may also cause trouble (zappos is 50k which isn't too bad), so you can try the usual tricks, e.g. randomly offset or rotate the input images slightly.
